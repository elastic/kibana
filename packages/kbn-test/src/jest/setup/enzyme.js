/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * and the Server Side Public License, v 1; you may not use this file except in
 * compliance with, at your election, the Elastic License or the Server Side
 * Public License, v 1.
 */

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
