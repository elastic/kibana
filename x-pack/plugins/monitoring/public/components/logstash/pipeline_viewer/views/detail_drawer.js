/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React from 'react';
import { last } from 'lodash';
import {
  EuiTitle,
  EuiText,
  EuiFlyout,
  EuiFlyoutHeader,
  EuiFlyoutBody,
  EuiTable,
  EuiTableBody,
  EuiTableRow,
  EuiTableRowCell,
  EuiCodeBlock,
  EuiFlexGroup,
  EuiFlexItem,
  EuiButtonIcon,
  EuiSpacer
} from '@elastic/eui';
import { Sparkline } from '../../../sparkline';
import { formatMetric } from '../../../../lib/format_number';
import { LOGSTASH } from '../../../../../common/constants';

function renderIcon(vertex) {
  return (
    <img
      src={vertex.icon}
      width={LOGSTASH.PIPELINE_VIEWER.ICON.WIDTH_PX}
      height={LOGSTASH.PIPELINE_VIEWER.ICON.HEIGHT_PX}
      className="lspvDetailDrawerIcon"
    />
  );
}

function renderPluginBasicStats(vertex, timeseriesTooltipXValueFormatter) {

  const eventsLatencyValueFormatter = (value) => formatMetric(value, '0.[00]a', 'ms/e');
  const latestEventsLatency =
    Array.isArray(vertex.stats.millis_per_event.data) && vertex.stats.millis_per_event.data.length > 0
      ? last(vertex.stats.millis_per_event.data)[1]
      : null;
  const eventsLatencyRow = vertex.pluginType === 'input'
    ? null
    : (
      <EuiTableRow key="events_latency">
        <EuiTableRowCell>Events Latency</EuiTableRowCell>
        <EuiTableRowCell>
          <div className="lspvDetailDrawerSparklineContainer">
            <Sparkline
              series={vertex.stats.millis_per_event.data}
              options={{ xaxis: vertex.stats.millis_per_event.timeRange }}
              tooltip={{
                enabled: true,
                xValueFormatter: timeseriesTooltipXValueFormatter,
                yValueFormatter: eventsLatencyValueFormatter
              }}
            />
          </div>
        </EuiTableRowCell>
        <EuiTableRowCell>
          { eventsLatencyValueFormatter(latestEventsLatency) }
        </EuiTableRowCell>
      </EuiTableRow>
    );

  const eventsOutRateValueFormatter = (value) => formatMetric(value, '0.[0]a', 'e/s');
  const eventsOutRateRow = (
    <EuiTableRow key="events_out_rate">
      <EuiTableRowCell>Events Emitted Rate</EuiTableRowCell>
      <EuiTableRowCell>
        <div className="lspvDetailDrawerSparklineContainer">
          <Sparkline
            series={vertex.eventsPerSecond.data}
            options={{ xaxis: vertex.eventsPerSecond.timeRange }}
            tooltip={{
              enabled: true,
              xValueFormatter: timeseriesTooltipXValueFormatter,
              yValueFormatter: eventsOutRateValueFormatter
            }}
          />
        </div>
      </EuiTableRowCell>
      <EuiTableRowCell>
        { eventsOutRateValueFormatter(vertex.latestEventsPerSecond) }
      </EuiTableRowCell>
    </EuiTableRow>
  );

  const eventsInValueFormatter = (value) => formatMetric(value, '0a', 'events');
  const latestEventsIn =
    Array.isArray(vertex.stats.events_in.data) && vertex.stats.events_in.data.length > 0
      ? last(vertex.stats.events_in.data)[1]
      : null;

  const eventsInRow = vertex.pluginType === 'input'
    ? null
    : (
      <EuiTableRow key="events_in">
        <EuiTableRowCell>Events Received</EuiTableRowCell>
        <EuiTableRowCell>
          <div className="lspvDetailDrawerSparklineContainer">
            <Sparkline
              series={vertex.stats.events_in.data}
              options={{ xaxis: vertex.stats.events_in.timeRange }}
              tooltip={{
                enabled: true,
                xValueFormatter: timeseriesTooltipXValueFormatter,
                yValueFormatter: eventsInValueFormatter
              }}
            />
          </div>
        </EuiTableRowCell>
        <EuiTableRowCell>
          { eventsInValueFormatter(latestEventsIn) }
        </EuiTableRowCell>
      </EuiTableRow>
    );

  const eventsOutValueFormatter = eventsInValueFormatter;
  const latestEventsOut =
    Array.isArray(vertex.stats.events_out.data) && vertex.stats.events_out.data.length > 0
      ? last(vertex.stats.events_out.data)[1]
      : null;
  const eventsOutRow = (
    <EuiTableRow key="events_out">
      <EuiTableRowCell>Events Emitted</EuiTableRowCell>
      <EuiTableRowCell>
        <div className="lspvDetailDrawerSparklineContainer">
          <Sparkline
            series={vertex.stats.events_out.data}
            options={{ xaxis: vertex.stats.events_out.timeRange }}
            tooltip={{
              enabled: true,
              xValueFormatter: timeseriesTooltipXValueFormatter,
              yValueFormatter: eventsOutValueFormatter
            }}
          />
        </div>
      </EuiTableRowCell>
      <EuiTableRowCell>
        { eventsOutValueFormatter(latestEventsOut) }
      </EuiTableRowCell>
    </EuiTableRow>
  );

  return (
    <EuiTable>
      <EuiTableBody>
        { eventsLatencyRow }
        { eventsOutRateRow }
        { eventsInRow }
        { eventsOutRow }
      </EuiTableBody>
    </EuiTable>
  );
}

function renderIfBasicStats(_vertex) {
  return (
    <p>There are currently no metrics to show for this if condition.</p>
  );
}

function renderQueueBasicStats(_vertex) {
  return (
    <p>There are currently no metrics to show for the queue.</p>
  );
}

function renderBasicStats(vertex, timeseriesTooltipXValueFormatter) {
  switch (vertex.typeString) {
    case 'plugin':
      return renderPluginBasicStats(vertex, timeseriesTooltipXValueFormatter);
      break;
    case 'if':
      return renderIfBasicStats(vertex, timeseriesTooltipXValueFormatter);
      break;
    case 'queue':
      return renderQueueBasicStats(vertex, timeseriesTooltipXValueFormatter);
      break;
  }
}

function renderPluginBasicInfo(vertex) {
  if (vertex.hasExplicitId) {
    return (
      <p>This {vertex.typeString}&#39;s ID is <strong>{ vertex.id }</strong>.</p>
    );
  }

  return (
    <div>
      <p>
        This {vertex.typeString} does not have an ID explicitly specified. Specifying an ID allows you to track differences
        across pipeline changes. You can explicitly specify an ID for this plugin like so:
      </p>
      <EuiCodeBlock>
        {vertex.name} {`{
  id => "mySpecialId"
}`}
      </EuiCodeBlock>
      <EuiSpacer />
    </div>
  );
}

function renderIfBasicInfo(vertex) {
  const ifCode = `if (${vertex.subtitle.complete}) {
  ...
}`;

  return (
    <div>
      <p>
        This is a conditional statement in your pipeline.
      </p>
      <EuiCodeBlock>{ ifCode }</EuiCodeBlock>
      <EuiSpacer />
    </div>
  );
}

function renderQueueBasicInfo(_vertex) {
  return (
    <p>
      This is an internal structure used by Logstash to buffer events between
      inputs and the rest of the pipeline.
    </p>
  );
}

function renderBasicInfo(vertex) {
  switch (vertex.typeString) {
    case 'plugin':
      return renderPluginBasicInfo(vertex);
      break;
    case 'if':
      return renderIfBasicInfo(vertex);
      break;
    case 'queue':
      return renderQueueBasicInfo(vertex);
      break;
  }
}

function renderTitle(vertex) {
  switch (vertex.typeString) {
    case 'plugin':
      return `${vertex.title} ${vertex.pluginType}`;
      break;
    case 'if':
    case 'queue':
      return vertex.title;
      break;
  }
}

export function DetailDrawer({ vertex, onHide, timeseriesTooltipXValueFormatter }) {
  return (
    <EuiFlyout
      size="s"
      onClose={onHide}
    >
      <EuiFlyoutHeader>
        <EuiFlexGroup
          alignItems="center"
        >
          <EuiFlexItem>
            <EuiTitle>
              <h2>{ renderIcon(vertex) }{ renderTitle(vertex) }</h2>
            </EuiTitle>
          </EuiFlexItem>
          <EuiFlexItem grow={false}>
            <EuiButtonIcon
              onClick={onHide}
              iconType="cross"
              aria-label="Close"
            />
          </EuiFlexItem>
        </EuiFlexGroup>
      </EuiFlyoutHeader>
      <EuiFlyoutBody>
        <EuiText>
          { renderBasicInfo(vertex) }
          { renderBasicStats(vertex, timeseriesTooltipXValueFormatter) }
        </EuiText>
      </EuiFlyoutBody>
    </EuiFlyout>
  );
}
