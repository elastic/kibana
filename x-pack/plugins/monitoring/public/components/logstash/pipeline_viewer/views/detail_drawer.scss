// SASSTODO: Why is this width here?
.lspvDetailDrawerSparklineContainer {
  width: 7vw;
}
