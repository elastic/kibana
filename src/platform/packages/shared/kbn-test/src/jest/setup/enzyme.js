/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the "Elastic License
 * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
 * Public License v 1"; you may not use this file except in compliance with, at
 * your election, the "Elastic License 2.0", the "GNU Affero General Public
 * License v3.0 only", or the "Server Side Public License, v 1".
 */

import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });

/* eslint-env jest */

/**
 * This is a workaround to fix snapshot serialization of emotion classes when rendering React@18 using `import { render } from 'enzyme'`
 * With React@18 emotion uses `useInsertionEffect` to insert styles into the DOM, which enzyme `render` does not trigger because it is using ReactDomServer.renderToString.
 * With React@17 emotion fell back to sync cb, so it was working with enzyme `render`.
 * Without those styles in DOM the custom snapshot serializer is not able to replace the emotion class names.
 * This workaround ensures a fake emotion style tag is present in the DOM before rendering the component with enzyme making the snapshot serializer work.
 */
function mockEnsureEmotionStyleTag() {
  if (!document.head.querySelector('style[data-emotion]')) {
    const style = document.createElement('style');
    style.setAttribute('data-emotion', 'css');
    document.head.appendChild(style);
  }
}

jest.mock('enzyme', () => {
  const actual = jest.requireActual('enzyme');
  return {
    ...actual,
    render: (node, options) => {
      mockEnsureEmotionStyleTag();
      return actual.render(node, options);
    },
  };
});
