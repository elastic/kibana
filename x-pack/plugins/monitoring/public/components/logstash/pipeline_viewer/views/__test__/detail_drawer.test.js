/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React from 'react';
import { DetailDrawer } from '../detail_drawer';
import { shallow } from 'enzyme';

describe('DetailDrawer component', () => {
  let onHide;
  let timeseriesTooltipXValueFormatter;

  beforeEach(() => {
    onHide = jest.fn();
    timeseriesTooltipXValueFormatter = jest.fn();
  });

  test('shows vertex title', () => {
    const vertex = {
      title: 'grok',
    };

    const component = <DetailDrawer vertex={vertex} onHide={onHide} />;
    const renderedComponent = shallow(component);
    expect(renderedComponent).toMatchSnapshot();
  });

  describe('Plugin vertices', () => {
    describe('Plugin has explicit ID', () => {
      test('shows basic info and stats for plugin, including explicit ID', () => {
        const vertex = {
          title: 'grok',
          typeString: 'plugin',
          pluginType: 'filter',
          hasExplicitId: true,
          id: 'parse_apache_logline',
          stats: {
            events_in: {
              data: [[1516131120000, 200], [1516131180000, 203]],
              timeRange: {
                min: 1516131138639,
                max: 1516135440463,
              },
            },
            events_out: {
              data: [[1516131120000, 199], [1516131180000, 200]],
              timeRange: {
                min: 1516131138639,
                max: 1516135440463,
              },
            },
            millis_per_event: {
              data: [[1516131120000, 0.21], [1516131180000, 0.23]],
              timeRange: {
                min: 1516131138639,
                max: 1516135440463,
              },
            },
          },
          eventsPerSecond: {
            data: [[1516131120000, 32], [1516131180000, 36]],
            timeRange: {
              min: 1516131138639,
              max: 1516135440463,
            },
          },
        };

        const component = (
          <DetailDrawer
            vertex={vertex}
            onHide={onHide}
            timeseriesTooltipXValueFormatter={timeseriesTooltipXValueFormatter}
          />
        );
        const renderedComponent = shallow(component);
        expect(renderedComponent).toMatchSnapshot();
      });
    });

    describe('Plugin does not have explicit ID', () => {
      test('shows basic info and stats for plugin, suggesting that user set explicit ID', () => {
        const vertex = {
          title: 'grok',
          typeString: 'plugin',
          pluginType: 'filter',
          hasExplicitId: false,
          id: 'foobarbazqux',
          stats: {
            events_in: {
              data: [[1516131120000, 200], [1516131180000, 203]],
              timeRange: {
                min: 1516131138639,
                max: 1516135440463,
              },
            },
            events_out: {
              data: [[1516131120000, 199], [1516131180000, 200]],
              timeRange: {
                min: 1516131138639,
                max: 1516135440463,
              },
            },
            millis_per_event: {
              data: [[1516131120000, 0.21], [1516131180000, 0.23]],
              timeRange: {
                min: 1516131138639,
                max: 1516135440463,
              },
            },
          },
          eventsPerSecond: {
            data: [[1516131120000, 32], [1516131180000, 36]],
            timeRange: {
              min: 1516131138639,
              max: 1516135440463,
            },
          },
        };

        const component = (
          <DetailDrawer
            vertex={vertex}
            onHide={onHide}
            timeseriesTooltipXValueFormatter={timeseriesTooltipXValueFormatter}
          />
        );
        const renderedComponent = shallow(component);
        expect(renderedComponent).toMatchSnapshot();
      });
    });
  });

  describe('If vertices', () => {
    test('shows basic info and no stats for if', () => {
      const vertex = {
        title: 'if',
        typeString: 'if',
        subtitle: '[type] == "apache_log"',
      };

      const component = (
        <DetailDrawer
          vertex={vertex}
          onHide={onHide}
          timeseriesTooltipXValueFormatter={timeseriesTooltipXValueFormatter}
        />
      );
      const renderedComponent = shallow(component);
      expect(renderedComponent).toMatchSnapshot();
    });
  });

  describe('Queue vertices', () => {
    test('shows basic info and no stats for queue', () => {
      const vertex = {
        title: 'queue',
        typeString: 'queue',
      };

      const component = (
        <DetailDrawer
          vertex={vertex}
          onHide={onHide}
          timeseriesTooltipXValueFormatter={timeseriesTooltipXValueFormatter}
        />
      );
      const renderedComponent = shallow(component);
      expect(renderedComponent).toMatchSnapshot();
    });
  });
});
