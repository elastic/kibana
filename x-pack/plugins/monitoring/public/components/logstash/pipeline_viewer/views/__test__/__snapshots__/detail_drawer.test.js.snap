// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailDrawer component If vertices shows basic info and no stats for if 1`] = `
<EuiFlyout
  closeButtonAriaLabel="Closes this dialog"
  hideCloseButton={false}
  maxWidth={false}
  onClose={[MockFunction]}
  ownFocus={false}
  size="s"
>
  <EuiFlyoutHeader
    hasBorder={false}
  >
    <EuiFlexGroup
      alignItems="baseline"
      component="div"
      direction="row"
      gutterSize="s"
      justifyContent="flexStart"
      responsive={true}
      wrap={false}
    >
      <EuiFlexItem
        component="div"
        grow={false}
      >
        <EuiIcon
          className="lspvDetailDrawerIcon"
          size="m"
        />
      </EuiFlexItem>
      <EuiFlexItem
        component="div"
        grow={true}
      >
        <EuiTitle
          size="m"
        >
          <h2>
            if
          </h2>
        </EuiTitle>
      </EuiFlexItem>
    </EuiFlexGroup>
  </EuiFlyoutHeader>
  <EuiFlyoutBody>
    <EuiText
      grow={true}
    >
      <div>
        <p>
          This is a conditional statement in your pipeline.
        </p>
        <EuiCodeBlock>
          if ([type] == "apache_log") {
  ...
}
        </EuiCodeBlock>
        <EuiSpacer
          size="l"
        />
      </div>
      <p>
        There are currently no metrics to show for this if condition.
      </p>
    </EuiText>
  </EuiFlyoutBody>
</EuiFlyout>
`;

exports[`DetailDrawer component Plugin vertices Plugin does not have explicit ID shows basic info and stats for plugin, suggesting that user set explicit ID 1`] = `
<EuiFlyout
  closeButtonAriaLabel="Closes this dialog"
  hideCloseButton={false}
  maxWidth={false}
  onClose={[MockFunction]}
  ownFocus={false}
  size="s"
>
  <EuiFlyoutHeader
    hasBorder={false}
  >
    <EuiFlexGroup
      alignItems="baseline"
      component="div"
      direction="row"
      gutterSize="s"
      justifyContent="flexStart"
      responsive={true}
      wrap={false}
    >
      <EuiFlexItem
        component="div"
        grow={false}
      >
        <EuiIcon
          className="lspvDetailDrawerIcon"
          size="m"
        />
      </EuiFlexItem>
      <EuiFlexItem
        component="div"
        grow={true}
      >
        <EuiTitle
          size="m"
        >
          <h2>
            grok filter
          </h2>
        </EuiTitle>
      </EuiFlexItem>
    </EuiFlexGroup>
  </EuiFlyoutHeader>
  <EuiFlyoutBody>
    <EuiText
      grow={true}
    >
      <div>
        <p>
          This 
          plugin
           does not have an ID explicitly specified. Specifying an ID allows you to track differences across pipeline changes. You can explicitly specify an ID for this plugin like so:
        </p>
        <EuiCodeBlock>
           
          {
  id =&gt; "mySpecialId"
}
        </EuiCodeBlock>
        <EuiSpacer
          size="l"
        />
      </div>
      <EuiTable
        responsive={true}
      >
        <EuiTableBody>
          <EuiTableRow
            key="events_latency"
          >
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              Events Latency
            </EuiTableRowCell>
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              <div
                className="lspvDetailDrawerSparklineContainer"
              >
                <Sparkline
                  options={
                    Object {
                      "xaxis": Object {
                        "max": 1516135440463,
                        "min": 1516131138639,
                      },
                    }
                  }
                  series={
                    Array [
                      Array [
                        1516131120000,
                        0.21,
                      ],
                      Array [
                        1516131180000,
                        0.23,
                      ],
                    ]
                  }
                  tooltip={
                    Object {
                      "enabled": true,
                      "xValueFormatter": [MockFunction],
                      "yValueFormatter": [Function],
                    }
                  }
                />
              </div>
            </EuiTableRowCell>
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              0.23 ms/e
            </EuiTableRowCell>
          </EuiTableRow>
          <EuiTableRow
            key="events_out_rate"
          >
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              Events Emitted Rate
            </EuiTableRowCell>
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              <div
                className="lspvDetailDrawerSparklineContainer"
              >
                <Sparkline
                  options={
                    Object {
                      "xaxis": Object {
                        "max": 1516135440463,
                        "min": 1516131138639,
                      },
                    }
                  }
                  series={
                    Array [
                      Array [
                        1516131120000,
                        32,
                      ],
                      Array [
                        1516131180000,
                        36,
                      ],
                    ]
                  }
                  tooltip={
                    Object {
                      "enabled": true,
                      "xValueFormatter": [MockFunction],
                      "yValueFormatter": [Function],
                    }
                  }
                />
              </div>
            </EuiTableRowCell>
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              N/A
            </EuiTableRowCell>
          </EuiTableRow>
          <EuiTableRow
            key="events_in"
          >
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              Events Received
            </EuiTableRowCell>
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              <div
                className="lspvDetailDrawerSparklineContainer"
              >
                <Sparkline
                  options={
                    Object {
                      "xaxis": Object {
                        "max": 1516135440463,
                        "min": 1516131138639,
                      },
                    }
                  }
                  series={
                    Array [
                      Array [
                        1516131120000,
                        200,
                      ],
                      Array [
                        1516131180000,
                        203,
                      ],
                    ]
                  }
                  tooltip={
                    Object {
                      "enabled": true,
                      "xValueFormatter": [MockFunction],
                      "yValueFormatter": [Function],
                    }
                  }
                />
              </div>
            </EuiTableRowCell>
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              203 events
            </EuiTableRowCell>
          </EuiTableRow>
          <EuiTableRow
            key="events_out"
          >
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              Events Emitted
            </EuiTableRowCell>
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              <div
                className="lspvDetailDrawerSparklineContainer"
              >
                <Sparkline
                  options={
                    Object {
                      "xaxis": Object {
                        "max": 1516135440463,
                        "min": 1516131138639,
                      },
                    }
                  }
                  series={
                    Array [
                      Array [
                        1516131120000,
                        199,
                      ],
                      Array [
                        1516131180000,
                        200,
                      ],
                    ]
                  }
                  tooltip={
                    Object {
                      "enabled": true,
                      "xValueFormatter": [MockFunction],
                      "yValueFormatter": [Function],
                    }
                  }
                />
              </div>
            </EuiTableRowCell>
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              200 events
            </EuiTableRowCell>
          </EuiTableRow>
        </EuiTableBody>
      </EuiTable>
    </EuiText>
  </EuiFlyoutBody>
</EuiFlyout>
`;

exports[`DetailDrawer component Plugin vertices Plugin has explicit ID shows basic info and stats for plugin, including explicit ID 1`] = `
<EuiFlyout
  closeButtonAriaLabel="Closes this dialog"
  hideCloseButton={false}
  maxWidth={false}
  onClose={[MockFunction]}
  ownFocus={false}
  size="s"
>
  <EuiFlyoutHeader
    hasBorder={false}
  >
    <EuiFlexGroup
      alignItems="baseline"
      component="div"
      direction="row"
      gutterSize="s"
      justifyContent="flexStart"
      responsive={true}
      wrap={false}
    >
      <EuiFlexItem
        component="div"
        grow={false}
      >
        <EuiIcon
          className="lspvDetailDrawerIcon"
          size="m"
        />
      </EuiFlexItem>
      <EuiFlexItem
        component="div"
        grow={true}
      >
        <EuiTitle
          size="m"
        >
          <h2>
            grok filter
          </h2>
        </EuiTitle>
      </EuiFlexItem>
    </EuiFlexGroup>
  </EuiFlyoutHeader>
  <EuiFlyoutBody>
    <EuiText
      grow={true}
    >
      <p>
        This 
        plugin
        's ID is 
        <EuiBadge
          color="default"
          iconSide="left"
        >
          parse_apache_logline
        </EuiBadge>
        .
      </p>
      <EuiTable
        responsive={true}
      >
        <EuiTableBody>
          <EuiTableRow
            key="events_latency"
          >
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              Events Latency
            </EuiTableRowCell>
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              <div
                className="lspvDetailDrawerSparklineContainer"
              >
                <Sparkline
                  options={
                    Object {
                      "xaxis": Object {
                        "max": 1516135440463,
                        "min": 1516131138639,
                      },
                    }
                  }
                  series={
                    Array [
                      Array [
                        1516131120000,
                        0.21,
                      ],
                      Array [
                        1516131180000,
                        0.23,
                      ],
                    ]
                  }
                  tooltip={
                    Object {
                      "enabled": true,
                      "xValueFormatter": [MockFunction],
                      "yValueFormatter": [Function],
                    }
                  }
                />
              </div>
            </EuiTableRowCell>
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              0.23 ms/e
            </EuiTableRowCell>
          </EuiTableRow>
          <EuiTableRow
            key="events_out_rate"
          >
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              Events Emitted Rate
            </EuiTableRowCell>
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              <div
                className="lspvDetailDrawerSparklineContainer"
              >
                <Sparkline
                  options={
                    Object {
                      "xaxis": Object {
                        "max": 1516135440463,
                        "min": 1516131138639,
                      },
                    }
                  }
                  series={
                    Array [
                      Array [
                        1516131120000,
                        32,
                      ],
                      Array [
                        1516131180000,
                        36,
                      ],
                    ]
                  }
                  tooltip={
                    Object {
                      "enabled": true,
                      "xValueFormatter": [MockFunction],
                      "yValueFormatter": [Function],
                    }
                  }
                />
              </div>
            </EuiTableRowCell>
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              N/A
            </EuiTableRowCell>
          </EuiTableRow>
          <EuiTableRow
            key="events_in"
          >
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              Events Received
            </EuiTableRowCell>
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              <div
                className="lspvDetailDrawerSparklineContainer"
              >
                <Sparkline
                  options={
                    Object {
                      "xaxis": Object {
                        "max": 1516135440463,
                        "min": 1516131138639,
                      },
                    }
                  }
                  series={
                    Array [
                      Array [
                        1516131120000,
                        200,
                      ],
                      Array [
                        1516131180000,
                        203,
                      ],
                    ]
                  }
                  tooltip={
                    Object {
                      "enabled": true,
                      "xValueFormatter": [MockFunction],
                      "yValueFormatter": [Function],
                    }
                  }
                />
              </div>
            </EuiTableRowCell>
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              203 events
            </EuiTableRowCell>
          </EuiTableRow>
          <EuiTableRow
            key="events_out"
          >
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              Events Emitted
            </EuiTableRowCell>
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              <div
                className="lspvDetailDrawerSparklineContainer"
              >
                <Sparkline
                  options={
                    Object {
                      "xaxis": Object {
                        "max": 1516135440463,
                        "min": 1516131138639,
                      },
                    }
                  }
                  series={
                    Array [
                      Array [
                        1516131120000,
                        199,
                      ],
                      Array [
                        1516131180000,
                        200,
                      ],
                    ]
                  }
                  tooltip={
                    Object {
                      "enabled": true,
                      "xValueFormatter": [MockFunction],
                      "yValueFormatter": [Function],
                    }
                  }
                />
              </div>
            </EuiTableRowCell>
            <EuiTableRowCell
              align="left"
              textOnly={true}
            >
              200 events
            </EuiTableRowCell>
          </EuiTableRow>
        </EuiTableBody>
      </EuiTable>
    </EuiText>
  </EuiFlyoutBody>
</EuiFlyout>
`;

exports[`DetailDrawer component Queue vertices shows basic info and no stats for queue 1`] = `
<EuiFlyout
  closeButtonAriaLabel="Closes this dialog"
  hideCloseButton={false}
  maxWidth={false}
  onClose={[MockFunction]}
  ownFocus={false}
  size="s"
>
  <EuiFlyoutHeader
    hasBorder={false}
  >
    <EuiFlexGroup
      alignItems="baseline"
      component="div"
      direction="row"
      gutterSize="s"
      justifyContent="flexStart"
      responsive={true}
      wrap={false}
    >
      <EuiFlexItem
        component="div"
        grow={false}
      >
        <EuiIcon
          className="lspvDetailDrawerIcon"
          size="m"
        />
      </EuiFlexItem>
      <EuiFlexItem
        component="div"
        grow={true}
      >
        <EuiTitle
          size="m"
        >
          <h2>
            queue
          </h2>
        </EuiTitle>
      </EuiFlexItem>
    </EuiFlexGroup>
  </EuiFlyoutHeader>
  <EuiFlyoutBody>
    <EuiText
      grow={true}
    >
      <p>
        This is an internal structure used by Logstash to buffer events between inputs and the rest of the pipeline.
      </p>
      <p>
        There are currently no metrics to show for the queue.
      </p>
    </EuiText>
  </EuiFlyoutBody>
</EuiFlyout>
`;

exports[`DetailDrawer component shows vertex title 1`] = `
<EuiFlyout
  closeButtonAriaLabel="Closes this dialog"
  hideCloseButton={false}
  maxWidth={false}
  onClose={[MockFunction]}
  ownFocus={false}
  size="s"
>
  <EuiFlyoutHeader
    hasBorder={false}
  >
    <EuiFlexGroup
      alignItems="baseline"
      component="div"
      direction="row"
      gutterSize="s"
      justifyContent="flexStart"
      responsive={true}
      wrap={false}
    >
      <EuiFlexItem
        component="div"
        grow={false}
      >
        <EuiIcon
          className="lspvDetailDrawerIcon"
          size="m"
        />
      </EuiFlexItem>
      <EuiFlexItem
        component="div"
        grow={true}
      >
        <EuiTitle
          size="m"
        >
          <h2 />
        </EuiTitle>
      </EuiFlexItem>
    </EuiFlexGroup>
  </EuiFlyoutHeader>
  <EuiFlyoutBody>
    <EuiText
      grow={true}
    />
  </EuiFlyoutBody>
</EuiFlyout>
`;
